internal/gcsx/syncer.go

```diff
@@ -219,8 +219,10 @@ func (os *syncer) SyncObject(
 		return
 	}
 
-	// Destroy the temp file.
-	content.Destroy()
+	// Destroy the temp file after syncing if the local file cache is disabled
+	if !localFileCache {
+		content.Destroy()
```

Comment:
This is no longer needed once it is rebased onto https://github.com/GoogleCloudPlatform/gcsfuse/pull/616